app/models/user.rb

```diff
@@ -3,4 +3,9 @@ class User < ApplicationRecord
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
+
+  validates :first_name, presence: true
+  validates :last_name, presence: true
+  validates :user_name, presence: true, uniqueness: true
+  validates :role, presence: true, inclusion: { in: ["member", "admin"] }
```

Comment:
Just FYI (not going to hold up merge but you can decide to change it later), you could remove `presence: true` here since `inclusion` validates presence by default!